### PR TITLE
Migration of xproj with name different from folder leaves ref to old project in solution

### LINKED ISSUE #5538: Migration of xproj with name different from folder leaves ref to old project in solution
## Steps to reproduce

Clone https://github.com/bleroy/core-imaging-playground/
Open a command-line on the /NetCore subdirectory
`dotnet migrate`

## Expected  behavior

Solution contains only reference to the migrated project. Old project files moved to backup.

## Actual behavior

Solution contains references to old (now gone) and new project. Old project has moved to backup.

## Environment data
`dotnet --info` output:

.NET Command Line Tools (1.0.0-rc4-004689)

Product Information:
 Version:            1.0.0-rc4-004689
 Commit SHA-1 hash:  c740fa8f43

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\1.0.0-rc4-004689

### PR DESCRIPTION
**Pull request by jgoshi**
Fixes #5538 

@piotrpMSFT @livarcocc @eerhardt @krwq @jonsequitur

### CODE REVIEW

**Comment by jgoshi** [comment]
test Windows_NT x64 Release Build

**Comment by jgoshi** [comment]
Seeing this intermittent failure:
Microsoft.DotNet.Tests.PackagedCommandTests.WhenToolAssetsFileIsInUseThenCLIRetriesLaunchingTheCommandForAtLeastOneSecond

**Comment by jgoshi** [comment]
test Windows_NT x86 Debug Build

**Comment by jgoshi** [comment]
test RHEL7.2 x64 Release Build

**Comment by jgoshi** [comment]
test Ubuntu16.04 x64 Debug Build

File: src/dotnet/commands/dotnet-migrate/MigrateCommand.cs [hunk 1]
```diff
@@ -224,6 +230,21 @@ private void AddProject(string slnPath, string csprojPath)
             addCommand.Execute();
         }
 
+        private void RemoveProject(string slnPath, string projPath)
```
Comment (eerhardt) [inline src/dotnet/commands/dotnet-migrate/MigrateCommand.cs]: This and `AddProject` appear to be almost identical.  Can they be refactored to share the logic?

File: src/dotnet/commands/dotnet-migrate/MigrateCommand.cs [hunk 2]
```diff
@@ -140,35 +141,35 @@ private void UpdateSolutionFile(MigrationReport migrationReport, SlnFile slnFile
...
-                    var csprojPath = Path.Combine(relativeReportPath, migratedProjectName);
-                    var solutionContainsCsprojPriorToMigration = slnFile.Projects
-                        .Where(p => p.FilePath == csprojPath)
+                    var xprojPath = Path.Combine(relativeReportPath, xprojFileName.Name);
+                    var solutionContainsXprojFileToRemove = slnFile.Projects
+                        .Where(p => p.FilePath == xprojPath)
                         .Any();
 
-                    if (!solutionContainsCsprojPriorToMigration)
+                    if (solutionContainsXprojFileToRemove)
```
Comment (eerhardt) [inline src/dotnet/commands/dotnet-migrate/MigrateCommand.cs]: Does "remove" fail if the sln file doesn't contain the project?  What if we just always tried to remove the project without needing to check if the project already existed in the .sln?

File: src/dotnet/commands/dotnet-migrate/MigrateCommand.cs (hunk 1 above)
Comment (jgoshi) [inline src/dotnet/commands/dotnet-migrate/MigrateCommand.cs]: Sure I can refactor

File: src/dotnet/commands/dotnet-migrate/MigrateCommand.cs (hunk 2 above)
Comment (jgoshi) [inline src/dotnet/commands/dotnet-migrate/MigrateCommand.cs]: I like having the check because then there's no need to spawn a new process unnecessarily.

Comment (eerhardt) [inline src/dotnet/commands/dotnet-migrate/MigrateCommand.cs]: makes sense.

**Comment by jgoshi** [comment]
@livarcocc @eerhardt When you have a chance could you let me know if this is ok to merge?

**Pull request merged**